asciidoc/modules/con_mg_changes-in-vertx-mqtt.adoc

```diff
@@ -0,0 +1,41 @@
+[id="changes-in-vertx-mqtt_{context}"]
+= Changes in {VertX} MQTT
+
+The following section describes the changes in {VertX} MQTT.
+
+== Some fluent methods in MQTT clients return future
+
+Some fluent methods in `MqttClient` class return `Future` instead of being fluent. For example, methods such as, `MqttClient.connect()`, `MqttClient.disconnect()`, `MqttClient.publish()` return future in {VertX} {4}.
+
+The following example shows the use of `publish()` method in {VertX} {3x} releases:
+
+----
+client
+   .publish("hello", Buffer.buffer("hello"), MqttQoS.EXACTLY_ONCE, false, false)
+   .publish("hello", Buffer.buffer("hello"), MqttQoS.AT_LEAST_ONCE, false, false);
+----
+
+The following example shows the use of `publish()` method in {VertX} {4} release:
+
+----
+client.publish("hello", Buffer.buffer("hello"), MqttQoS.EXACTLY_ONCE, false, false);
+client.publish("hello", Buffer.buffer("hello"), MqttQoS.AT_LEAST_ONCE, false, false);
+----
+
+== Mqtt will message returns a buffer
+
+The `MqttWill` data object gets and sets a string message as a buffer instead of a byte array. XXX
+
+== Removed the deprecated will and authorization methods from MQTT
+
+The following MQTT methods have been removed:
+
+|===
+|Removed methods|Replacing methods
+|`MqttWill.willMessage()`| `MqttWill.getWillMessage()`
+|`MqttWill.willTopic()`| `MqttWill.getWillTopic()   `
+|`MqttWill.willQos()`|`MqttWill.getWillQos()`
+|`MqttAuth.username()`|`MqttAuth.getUsername()`
+|`MqttAuth.password()`|`MqttAuth.getPassword()`
+|`MqttClientOptions.setKeepAliveTimeSeconds()`|`MqttClientOptions.setKeepAliveInterval()`XXX
+|===
```

Comment:
@vietj In the wiki page, the method is documented with parameter 4. Is that the default value? Do we need to say something about it?
https://github.com/vert-x3/wiki/wiki/4.0.0-Deprecations-and-breaking-changes#%EF%B8%8F-removed-deprecated-methods
***************************************************************************************************************
On MqttClientOptions

// 3.x
options.setKeepAliveTimeSeconds(4);

// 4.0
options.setKeepAliveInterval(4);

Reply:
there is no value by default for this setting, this is just an example.

Reply:
@vietj Thank you for the information. No changes required in doc then.